sound/soc/sof/intel/tgl.c

```diff
@@ -91,6 +91,8 @@ int sof_tgl_ops_init(struct snd_sof_dev *sdev)
 
 		ipc4_data->mtrace_type = SOF_IPC4_MTRACE_INTEL_CAVS_2;
 
+		ipc4_data->fw_context_save = true;
+
```

Comment:
Why not set this in `sof_ipc4_query_fw_configuration()` before starting to parse the received fw_config?
Otherwise APL/SKL/etc also needs to have this flag set to true for IPC4.

Reply:
@ujfalusi I don't think APL/SKL/ can support IPC4 since (1) no power support on zephyr side. I worked on power setting on zephyr side but don't find any support other than cavs2.5 (2) we don't add these old platform in our CI so the definitely can't work

Reply:
APL/SKL is not supported with SOF, only with reference fw, which supports context save/restore.